app/models/controlled_work_document_content.rb

```diff
@@ -0,0 +1,206 @@
+# This model presents the session data for a check in a way that enables ControlledWorkDocumentValueMappingService
+# to access all the values specified by any given "mapping". These mappings either specify a direct attribute
+# or a "CFE payload path".
+
+# In the former case, the attribute is either one that the `Check` superclass makes available (i.e. it's
+# an attribute that's available in the session, with Check's extra logic to make sure that it's
+# an attribute associated with a screen that is relevant to the check), or its an attribute that's defined
+# explicitly as a method below
+
+# In the latter case, the "path" specifies the location in the "api_response" hash that we expect to find
+# in the session, which contains all the information CFE returned when we asked it to perform an
+# eligibility calculation for this check
+class ControlledWorkDocumentContent < Check
+  include ActionView::Helpers::NumberHelper
+
+  def from_cfe_payload(path)
+    path_parts = path.split(".").map { _1.to_i.to_s == _1 ? _1.to_i : _1 }
+    value = session_data.dig("api_response", *path_parts)
+    format(value)
+  end
+
+  def from_attribute(attribute)
+    format(send(attribute))
+  end
+
+private
+
+  def format(value)
+    return unless value
+    return value unless value.is_a?(Numeric)
+
+    precision = value.round == value ? 0 : 2
+
+    number_with_precision(value, precision:, delimiter: ",")
+  end
+
+  def smod_assets?
+    house_in_dispute || vehicle_in_dispute || in_dispute.present?
+  end
+
+  def additional_property_in_dispute?
+    in_dispute.include? "property"
+  end
+
+  def savings_in_dispute?
+    in_dispute.include? "savings"
+  end
+
+  def investments_in_dispute?
+    in_dispute.include? "investments"
+  end
+
+  def valuables_in_dispute?
+    in_dispute.include? "valuables"
+  end
+
+  def main_home_percentage_owned
+    if joint_ownership
+      percentage_owned + joint_percentage_owned
+    else
+      percentage_owned || partner_percentage_owned
+    end
+  end
+
+  def smod_total
+    "TODO"
+  end
+
+  def non_smod_total
+    "TODO"
+  end
+
+  def additional_non_smod_properties_value
+    additional_non_smod_properties_sum "value"
+  end
+
+  def additional_non_smod_properties_mortgage
+    additional_non_smod_properties_sum "outstanding_mortgage"
+  end
+
+  def additional_non_smod_properties_percentage_owned
```

Comment:
My concern with returning a blank is that we will forget about this method and it will look like a bug to a user, i.e they enter something relevant that is not subsequently included on the form. Would it be better to highlight somewhere on the form that we cannot display due to [reason], instead of appearing to ignore it

Reply:
That's a really good idea. Maybe it should say "Unknown" or something in the field?